src/loaders/compute_tools/checkm2/checkm2.py

```diff
@@ -69,14 +66,23 @@ def _run_checkm2(
         fatal_tuples.append(fatal_tuple)
     write_fatal_tuples_to_dict(fatal_tuples, output_dir)
 
-    metadata = {'tool': 'checkm2',
+    end_time = time.time()
+    run_time = end_time - start
+    print(f"Used {round(run_time / 60, 2)} minutes to execute checkM2 predict "
+          + f"for {size} genomes"
+          )
+
+    metadata = {'tool_name': 'checkm2',
                 'version': '1.0.1',
                 'command': command,
                 "reference_db": {
                     "version": None,
                     "comment": "diamond_db, ver unknown",
                     },
-                'ids_to_files': ids_to_files}
+                'ids_to_files': make_json_serializable(ids_to_files),
+                'run_time': round(run_time, 2),
+                'batch_size': size,
+                }
```

Comment:
Wouldn't it make more sense, now I think about it, to handle this outside the container? Then the code would live in one place rather than having to duplicate it for every container

Reply:
hmmm which block of code are you referring to? Generate the `metadata` outside of container? something like making `create_tool_metadata` takes `command, run_time, size, etc.`?

Reply:
Generate the metadata outside of container?

right. The running code should know all this stuff, so you could abstract it all out of the containers

Reply:
👍

Reply:
This is something that could probably be done later though, as I assume it'd be a bit lift

Reply:
👍 I moved everything to `create_tool_metadata`.

Reply:
Sorry, missed this.

I mean the metadata code doesn't run in the container at all, and is written by code after the container exits. Then you don't need to add it to every tool, it Just Works

Reply:
That probably won’t be a good idea. If we move this logic completely outside of container, we will need if statement for each tool for tool specific metadata. And For ‘command’ metadata, we also need to return then command.

Reply:
I don't think so, most of the additional metadata is either source file, data id, ref db version, or ids_to_files, which is either known or can be looked up by looking in the tool directory. The only anomaly is mash, and for that there could be an method in each tool dir, that if it exists, takes the input args and returns the extra metadata.

For the command, the command = fn(input_args) so there could also be a method that does that translation that is used both in the container to start the executable and outside the container to generate the metadata.

Anyway, that's a different PR if it gets done at all.

Reply:
👍